cli/app.go

```diff
@@ -490,6 +491,9 @@ func (c *App) runAppWithContext(command *kingpin.CmdClause, cb func(ctx context.
 		return errors.Wrap(err, "unable to start metrics")
 	}
 
+	s := profilelog.MaybeStart(ctx)
+	defer s.Stop(ctx)
```

Comment:
This could be explicitly called below before calling `observability.stopMetrics()` instead of `defer`ing.